### PR TITLE
Xxs attack

### PR DESCRIPTION
**Pull request by siddhantgosavi**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by siddhantgosavi** [comment]
Hi @bmoore-msft , could you please review the pull request.

**Comment by girishjaju** [comment]
Hi @bmoore-msft , please review and authorize the pull request.

**Comment by bmoore-msft** [comment]
I suspect this is going to require same changes as in #4909 so lmk when the updates are here as well

**Comment by siddhantgosavi** [comment]
hi @bmoore-msft , I've made changes suggested in #4909 here as well. please review.

File: 101-XSS-Attack-Prevention/nested/microsoft.loganalytics/workspaces.json [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        "type": "Microsoft.OperationalInsights/workspaces/datasources",
+        "kind": "AzureActivityLog",
+        "name": "[concat(parameters('omsWorkspaceName'), '/', subscription().subscriptionId)]",
+        "apiVersion": "2015-11-01-preview",
+        "location": "[parameters('location')]",
+        "dependsOn": [
+          "[parameters('omsWorkspaceName')]"
+        ],
+        "properties": {
+          "linkedResourceId": "[concat(subscription().Id, '/providers/microsoft.insights/eventTypes/management')]"
```
Comment (bmoore-msft) [inline 101-XSS-Attack-Prevention/nested/microsoft.loganalytics/workspaces.json]: use resourceId here...

Comment (siddhantgosavi) [reply 101-XSS-Attack-Prevention/nested/microsoft.loganalytics/workspaces.json]: hi @bmoore-msft , we could not figure out how can we use resourceID function to get this resource id. I looked after some references in the quickstart itself to find out that the same method was used to get it.
https://github.com/Azure/azure-quickstart-templates/blob/master/azmgmt-demo/nestedtemplates/omsWorkspace.json
https://github.com/Azure/azure-quickstart-templates/blob/master/oms-all-deploy/nestedtemplates/omsWorkspace.json
please let us know if you have any suggestion on how to use it.

Comment (bmoore-msft) [reply 101-XSS-Attack-Prevention/nested/microsoft.loganalytics/workspaces.json]: resourceId('Microsoft.Insights/eventTypes', 'management')

Comment (siddhantgosavi) [reply 101-XSS-Attack-Prevention/nested/microsoft.loganalytics/workspaces.json]: i've already tried this solution earlier. it throws bad request error. i tried passing subscription id to the function as well. the resourceID function returns id with resourcegroup name in it as well. this property is not expecting that value.
image

Comment (siddhantgosavi) [reply 101-XSS-Attack-Prevention/nested/microsoft.loganalytics/workspaces.json]: Hi @bmoore-msft , I've raised a separate PR including both sql injection and xss scenario. This one can be closed.
the resourceId('Microsoft.Insights/eventTypes', 'management') function is not working and i had to run it as done in the following example.
https://github.com/Azure/azure-quickstart-templates/blob/master/azmgmt-demo/nestedtemplates/omsWorkspace.json
https://github.com/Azure/azure-quickstart-templates/blob/master/oms-all-deploy/nestedtemplates/omsWorkspace.json

**Pull request closed without merging**